packages/zero-solid/src/use-zero.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  batch,
+  createContext,
+  createMemo,
+  onCleanup,
+  useContext,
+  type JSX,
+} from 'solid-js';
+import {
+  Zero,
+  type CustomMutatorDefs,
+  type Schema,
+  type ZeroOptions,
+} from '../../zero/src/zero.ts';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention, @typescript-eslint/no-explicit-any
+const ZeroContext = createContext<(() => Zero<any, any>) | undefined>(
+  undefined,
+);
+
+export function createZero<S extends Schema, MD extends CustomMutatorDefs<S>>(
+  options: ZeroOptions<S, MD>,
+): Zero<S, MD> {
+  const opts = {
+    ...options,
+    batchViewUpdates: batch,
+  };
+  return new Zero(opts);
+}
+
+export function useZero<
+  S extends Schema,
+  MD extends CustomMutatorDefs<S> | undefined = undefined,
+>(): () => Zero<S, MD> | undefined {
+  const zero = useContext(ZeroContext);
+
+  // TODO: uncomment when we require ZeroProvider in a future release.
+  // if (zero === undefined) {
+  //   throw new Error('useZero must be used within a ZeroProvider');
+  // }
+  return zero ?? (() => undefined);
+}
+
+export function createUseZero<
+  S extends Schema,
+  MD extends CustomMutatorDefs<S> | undefined = undefined,
+>() {
+  return () => useZero<S, MD>();
+}
+
+export function ZeroProvider<
+  S extends Schema,
+  MD extends CustomMutatorDefs<S> | undefined = undefined,
+>(props: {
+  children: JSX.Element;
+  zeroSignal: () => ZeroOptions<S, MD> | {zero: Zero<S, MD>};
+}) {
+  const zero = createMemo(() => {
+    const z = props.zeroSignal();
+    if ('zero' in z) {
+      return z.zero;
+    }
+    const createdZero = new Zero({
+      ...z,
+      batchViewUpdates: batch,
+    });
+    onCleanup(() => createdZero.close());
+    return createdZero;
+  });
+
+  return ZeroContext.Provider({
+    value: zero,
+    get children() {
+      return props.children;
+    },
+  });
```

Comment:
```suggestion
  return <ZeroContext.Provider value={zero}>
    {props.children}
  </ZeroContext.Provider>
```
You should be able to do just this

Reply:
We need to avoid using JSX here as esbuild doesn't seem able to build a library that uses both React JSX and Solid JSX (at least we couldn't figure it out).